util/collections/src/garbage/atomic/counted.rs

```diff
@@ -58,43 +55,36 @@ impl GcCounter {
 	/// Increments the value in a specific slot.
 	pub fn increment(&self, current_time: u64, amount: u64) {
 		let slot_timestamp = current_time / self.gc_slot_duration.get();
-		let slot = slot_timestamp % self.num_slots;
+		let pass_number = slot_timestamp / (self.num_slots / 2); // Determine if we're in the first or second pass
+		let base_slot = slot_timestamp % (self.num_slots / 2); // Calculate the base slot index within the first pass
+
+		// Calculate the actual slot index by adding the pass offset (0 or 1 times num_slots / 2)
+		let slot = base_slot + (pass_number % 2) * (self.num_slots / 2);
```

Comment:
I don't think this fundamentally solves the problem with `atomic::GcCounter`, though. A thread could still be scheduled off for two (or more) passes and contest the same slot with a different timestamp.

Reply:
Yes, but correct usage would mean this should not happen in a way that continues to sum. That is, this should have been garbage collected.